libs/blocks/global-navigation/blocks/profile/signIn.js

```diff
@@ -0,0 +1,30 @@
+const decorateSignIn = async ({ blockEl, decoratedEl, toFragment, getPlaceholder }) => {
+  const dropDown = blockEl.querySelector(':scope > div:nth-child(2)');
+  const signInLabel = await getPlaceholder('sign-in').then(({ value }) => value);
+  if (!dropDown) {
+    const signIn = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signin">${signInLabel}</a>`;
+    signIn.addEventListener('click', (e) => {
+      e.preventDefault();
+      window.adobeIMS.signIn();
+    });
+    return signIn;
+  }
+
+  const signInWithDropDown = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signin" role="button" aria-expanded="false" aria-controls="navmenu-profile" daa-lh="header|Close" aria-haspopup="true">${signInLabel}</a>`;
+  dropDown.classList.add('feds-signin-dropdown');
+  decoratedEl.insertAdjacentElement('beforeend', dropDown);
+
+  // TODO we don't have a good way of adding attributes to links
+  const dropDownSignIn = dropDown.querySelector('[href="https://adobe.com?sign-in=true"]');
```

Comment:
I'm not sure I follow this selector; why are we using `?sign-in=true`?

Reply:
We use it to know which link is used for signing in (and add an eventListener for IMS) instead of following the link.

There's a TODO above `// TODO we don't have a good way of adding attributes to links`

Reply:
I saw the todo, but did not understand the use-case; now I get it, thanks!